packages/gasket-plugin-docs-graph/plugin.js

```diff
@@ -0,0 +1,41 @@
+/* eslint no-sync: 0 */
+const path = require('path');
+const fs = require('fs');
+const write = require('util').promisify(fs.writeFile);
+
+module.exports = {
+  name: 'docs-graph',
+  hooks: {
+    async docsGenerate(gasket, docsConfigSet) {
+      const { docsRoot } = docsConfigSet;
+      const targetRoot = path.join(docsRoot, 'generated-docs');
+      const target = path.join(targetRoot, 'mermaid.md');
+
+      let graph = 'graph LR;\n';
+      docsConfigSet.lifecycles.forEach(lifecycle => {
+        const name = lifecycle.name;
+        const from = lifecycle.parent || lifecycle.after || lifecycle.command || lifecycle.from;
+        const arrow = lifecycle.method ? `-- ${lifecycle.method} -->` : '-->';
+        if (name !== from) {
+          graph += `${from} ${arrow} ${name};\n`;
+        }
+      });
+
+      graph = graph.replace(/@/g, '');
+      const content = '```mermaid\n' + graph + '```';
+
+      if (!fs.existsSync(targetRoot)) {
```

Comment:
Isn't this check unnecessary since this lifecycle runs after collateFiles?

Reply:
`generated-files` is a new directory specifically made for this `docsGenerate` hook. It would otherwise not be generated at all, throwing an error when trying to write to it.